Add diagnostic items for Clippy

samueltardieu opened the pull request:
Clippy still uses some paths to access items from the standard library. Adding the missing diagnostic items allows removing the last remaining paths.

Closes rust-lang/rust-clippy#5393

samueltardieu committed:
Add diagnostic items for Clippy

samueltardieu committed:
clippy: replace path uses by diagnostic items

Urgau reviewed (approved):
r=me on the compiler side

Manishearth commented:
@bors r+

bors commented:
:pushpin: Commit 07338a40de90d64155ca25bb58fc24cf9d6b0caa has been approved by `Manishearth`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Manishearth 07338a40de90d64155ca25bb58fc24cf9d6b0caa -->
<!-- homu: {"type":"Approved","sha":"07338a40de90d64155ca25bb58fc24cf9d6b0caa","approver":"Manishearth","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

flip1995 commented:
It does. Sorry for the ping 😅 r? Manishearth

Urgau commented:
@bors r-

samueltardieu committed:
Bless UI tests: do not print alternate implementors of `Step`

Alternative candidates for a trait implementation are not printed when
the trait has a diagnostic name, to avoid printing alternatives for
common stdlib traits such as `Copy` or `Debug`. However, this affects
all traits for which a diagnostic item is added.

Here, the list of alternatives candidates for `Step` does not seem
useful, and `Step` is unstable, so this will not be missed.

Urgau commented:
I think it's fine.

@bors r=Manishearth,Urgau rollup

bors commented:
:pushpin: Commit 890ade5ae85300cc96b1f948d4ac2d756cc67ccd has been approved by `Manishearth,Urgau`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Manishearth,Urgau 890ade5ae85300cc96b1f948d4ac2d756cc67ccd -->
<!-- homu: {"type":"Approved","sha":"890ade5ae85300cc96b1f948d4ac2d756cc67ccd","approver":"Manishearth,Urgau","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
